ipc/provider/src/lib.rs

```diff
@@ -4,8 +4,407 @@
+                config::subnet::SubnetConfig::Fevm(_) => {
+                    let manager = Box::new(
+                        FevmSubnetManager::from_subnet_with_wallet_store(
+                            subnet,
+                            self.evm_keystore.clone(),
+                            self.fvm_wallet.clone(),
+                        )
+                        .ok()?,
+                    );
+                    Some(Connection {
+                        manager,
+                        subnet: subnet.clone(),
+                    })
+                }
+            },
+            None => None,
+        }
+    }
+
+    /// Set the default account for the provider
+    pub fn with_sender(&mut self, from: Address) {
+        self.sender = Some(from);
+    }
+
+    fn call_sender(&self, from: Option<Address>) -> anyhow::Result<Address> {
+        if from.is_none() {
+            // get default account
+            if self.sender.is_none() {
+                return Err(anyhow!(
+                    "default account not found for provider. Use `with_sender` to set it up."
+                ));
+            } else {
+                return Ok(self.sender.unwrap());
+            }
+        }
+        Ok(from.unwrap())
+    }
+
+    fn check_subnet(&self, subnet: &config::Subnet) -> anyhow::Result<()> {
+        match &subnet.config {
+            config::subnet::SubnetConfig::Fvm(config) => {
+                if config.auth_token.is_none() {
+                    log::error!("subnet {:?} does not have auth token", subnet.id);
+                    return Err(anyhow!("Internal server error"));
+                }
+            }
+            config::subnet::SubnetConfig::Fevm(_) => {
+                // TODO: add more checks later
+            }
+        }
+        Ok(())
+    }
+}
+
+/// IpcProvider spawns a daemon-less client to interact with IPC subnets.
+///
+/// At this point the provider assumes that the user providers a `config.toml`
+/// with the subnet configuration. This has been inherited by the daemon
+/// configuration and will be slowly deprecated.
+impl IpcProvider {
+    // FIXME: Once the arguments for subnet creation are stabilized,
+    // use a SubnetOpts struct to provide the creation arguments and
+    // remove this allow
+    #[allow(clippy::too_many_arguments)]
+    pub async fn create_subnet(
+        &self,
+        from: Option<Address>,
+        parent: &SubnetID,
+        subnet_name: String,
+        min_validators: u64,
+        min_validator_stake: TokenAmount,
+        bottomup_check_period: ChainEpoch,
+        topdown_check_period: ChainEpoch,
+    ) -> anyhow::Result<Address> {
+        let conn = match self.connection(parent) {
+            None => return Err(anyhow!("target parent subnet not found")),
+            Some(conn) => conn,
+        };
+
+        let subnet_config = conn.subnet();
+        self.check_subnet(subnet_config)?;
+
+        let constructor_params = ConstructParams {
```

Comment:
btw, should we update the params here, a lot of the fields are outdated.

Reply:
Should we update once we integrate with the contracts? I am still testing against the current version of contracts currently included in `main` and in the agent.

Reply:
I am also thinking about propagating the compilation of actors in compile time that Akosh is using in Fendermint here so we don't have to worry about version mismatches.